nautobot/ipam/filters.py

```diff
@@ -388,34 +394,44 @@ class IPAddressFilterSet(NautobotFilterSet, TenancyFilterSet, StatusModelFilterS
         to_field_name="rd",
         label="VRF (RD)",
     )
-    present_in_vrf_id = django_filters.ModelChoiceFilter(
-        queryset=VRF.objects.all(), method="filter_present_in_vrf", label="VRF"
-    )
-    present_in_vrf = django_filters.ModelChoiceFilter(
-        queryset=VRF.objects.all(),
-        method="filter_present_in_vrf",
-        to_field_name="rd",
-        label="VRF (RD)",
-    )
-    device = MultiValueCharFilter(
-        method="filter_device",
-        field_name="name",
-        label="Device (name)",
-    )
-    device_id = MultiValueCharFilter(
-        method="filter_device",
-        field_name="pk",
-        label="Device (ID)",
-    )
-    virtual_machine = MultiValueCharFilter(
-        method="filter_virtual_machine",
-        field_name="name",
-        label="Virtual machine (name)",
-    )
-    virtual_machine_id = MultiValueCharFilter(
-        method="filter_virtual_machine",
-        field_name="pk",
-        label="Virtual machine (ID)",
+    present_in_vrf_id = extend_schema_field(OpenApiTypes.UUID)(
+        django_filters.ModelChoiceFilter(queryset=VRF.objects.all(), method="filter_present_in_vrf", label="VRF")
+    )
+    present_in_vrf = extend_schema_field(str)(
+        django_filters.ModelChoiceFilter(
+            queryset=VRF.objects.all(),
+            method="filter_present_in_vrf",
+            to_field_name="rd",
+            label="VRF (RD)",
+        )
+    )
+    device = extend_schema_field(str)(
+        MultiValueCharFilter(
+            method="filter_device",
+            field_name="name",
+            label="Device (name)",
+        )
+    )
+    device_id = extend_schema_field(OpenApiTypes.UUID)(
+        MultiValueCharFilter(
+            method="filter_device",
+            field_name="pk",
+            label="Device (ID)",
+        )
+    )
+    virtual_machine = extend_schema_field(str)(
+        MultiValueCharFilter(
+            method="filter_virtual_machine",
+            field_name="name",
+            label="Virtual machine (name)",
+        )
+    )
+    virtual_machine_id = extend_schema_field(OpenApiTypes.UUID)(
+        MultiValueCharFilter(
+            method="filter_virtual_machine",
+            field_name="pk",
+            label="Virtual machine (ID)",
+        )
```

Comment:
Hmm is this something that's going to have to happen to every FilterSet? This kind of freaks me out. Is this perhaps something we could instead do at the filter field definition level or do by way more with metaprogramming?

Reply:
Probably related to https://github.com/tfranzel/drf-spectacular/issues/690?

Reply:
I've reworked this approach in d1a61ca3a1ae5374c32736664340657002013112 - as you suggested, refining the definition of our custom filter types (`MultiValue*Filter`) made it possible to get more accurate schema generation without needing to do a bunch of individual overrides. For `present_in_vrf` and `present_in_vrf_id` specifically, adding an appropriate `field_name` parameter on the `ModelChoiceFilter` definition seems to have fixed those.